### PR TITLE
[wasm] [sdb] Initial PR towards making some of sdb's code shareable.

### PR DESCRIPTION
**Pull request by kumpera**
This PR adds the new files, moves the enums / structs and a small bit of code to validate the change.

### CODE REVIEW

File: mono/mini/debugger-agent.c [hunk 1]
```diff
@@ -286,79 +287,6 @@ typedef enum {
...
-	EVENT_KIND_ASSEMBLY_LOAD = 8,
-	EVENT_KIND_ASSEMBLY_UNLOAD = 9,
-	EVENT_KIND_BREAKPOINT = 10,
-	EVENT_KIND_STEP = 11,
-	EVENT_KIND_TYPE_LOAD = 12,
-	EVENT_KIND_EXCEPTION = 13,
-	EVENT_KIND_KEEPALIVE = 14,
-	EVENT_KIND_USER_BREAK = 15,
-	EVENT_KIND_USER_LOG = 16
-} EventKind;
```
Comment (vargaz) [inline mono/mini/debugger-agent.c]: Does this have to move too, doesn't it belong to the debugger protocol ?

File: mono/mini/debugger-agent.c [hunk 2]
```diff
@@ -286,79 +287,6 @@ typedef enum {
...
-
-typedef enum {
-	SUSPEND_POLICY_NONE = 0,
-	SUSPEND_POLICY_EVENT_THREAD = 1,
-	SUSPEND_POLICY_ALL = 2
-} SuspendPolicy;
-
```
Comment (vargaz) [inline mono/mini/debugger-agent.c]: Same.

File: mono/mini/debugger-agent.c [hunk 3]
```diff
@@ -286,79 +287,6 @@ typedef enum {
...
-	ERR_NOT_SUSPENDED = 101,
-	ERR_INVALID_ARGUMENT = 102,
-	ERR_UNLOADED = 103,
-	ERR_NO_INVOCATION = 104,
-	ERR_ABSENT_INFORMATION = 105,
-	ERR_NO_SEQ_POINT_AT_IL_OFFSET = 106,
-	ERR_INVOKE_ABORTED = 107,
-	ERR_LOADER_ERROR = 200, /*XXX extend the protocol to pass this information down the pipe */
-} ErrorCode;
-
```
Comment (vargaz) [inline mono/mini/debugger-agent.c]: Same.

**Comment by lewurm** [comment]
is `mono/mini/mini-wasm-debugger.c` supposed to be an empty file?

File: mono/mini/debugger-agent.c (hunk 1 above)
Comment (kumpera) [reply mono/mini/debugger-agent.c]: The problem is that those 3 enums are used deep in the breakpoint/single step code.

Let me try to see how much would need to be shuffled around to move them back.

**Comment by kumpera** [comment]
@lewurm yes, this PR just add makefile scaffold and move a bit of code around to validate the change.

**Comment by kumpera** [comment]
@vargaz I moved SuspendPolicy and ErrorCode back to sdb.
EventKind will be quite some work as it's stored in EventRequest and that gets shared across.
I added a FIXME to later deal with it properly. I think that's an ok compromise for now.

**Comment by kumpera** [comment]
Last commit fixes the wasm breakage.

**Comment by kumpera** [comment]
@monojenkins build Linux i386

**Pull request merged**